src/WriteStream.h

```diff
@@ -88,30 +92,66 @@ class WriteStream {
     void writeNext(const Tuple tuple) {
         writeNextTuple(tuple.data);
     }
-    void writeNextTupleElement(std::ostream& destination, RamTypeAttribute type, RamDomain value) {
-        switch (type) {
-            case RamTypeAttribute::Symbol:
-                destination << symbolTable.unsafeResolve(value);
-                break;
-            case RamTypeAttribute::Signed:
-                destination << value;
-                break;
-            case RamTypeAttribute::Unsigned:
-                destination << ramBitCast<RamUnsigned>(value);
-                break;
-            case RamTypeAttribute::Float:
-                destination << ramBitCast<RamFloat>(value);
-                break;
-            case RamTypeAttribute::Record:
-                assert(false && "Record writing is not supported");
+
+    void outputRecord(std::ostream& destination, const RamDomain value, const std::string& name) {
+        Json recordInfo = types["records"][name];
+
+        // Check if record type information are present
+        if (recordInfo.is_null()) {
+            std::cerr << "Missing record type information: " << name << std::endl;
+            abort();
```

Comment:
Could this be a throw instead? Or perhaps replace the whole if statement with an assert.